src/back/spv/writer.rs

```diff
@@ -724,9 +769,13 @@ impl Writer {
     fn parse_type_declaration(
         &mut self,
         arena: &crate::Arena<crate::Type>,
-        handle: crate::Handle<crate::Type>,
+        lookup_type: &LookupType<crate::Type>,
     ) -> (Instruction, Word) {
-        let ty = &arena[handle];
+        let ty = match lookup_type {
+            LookupType::Handle(handle) => &arena[*handle],
+            LookupType::Local(value) => &value,
+        };
```

Comment:
This feels pretty wrong to do it like this. Would it be better to let both LookupTypes implement a trait, where I can retrieve type T from, instead of matching and trying to get the Type (in this case).

Reply:
it's fine if we don't do this many times :)